compiler/fir/checkers/gen/org/jetbrains/kotlin/fir/analysis/checkers/declaration/DeclarationCheckers.kt

```diff
@@ -37,6 +37,6 @@ abstract class DeclarationCheckers {
     @CheckersComponentInternal internal val allPropertyCheckers: Set<FirPropertyChecker> get() = propertyCheckers + allMemberDeclarationCheckers
     @CheckersComponentInternal internal val allClassCheckers: Set<FirClassChecker> get() = classCheckers + allBasicDeclarationCheckers
     @CheckersComponentInternal internal val allRegularClassCheckers: Set<FirRegularClassChecker> get() = regularClassCheckers + allMemberDeclarationCheckers + allClassCheckers
-    @CheckersComponentInternal internal val allConstructorCheckers: Set<FirConstructorChecker> get() = constructorCheckers + allFunctionCheckers
+    @CheckersComponentInternal internal val allConstructorCheckers: Set<FirConstructorChecker> get() = constructorCheckers + allFunctionCheckers + memberDeclarationCheckers
```

Comment:
I don't think I understand this change... Rather, should we change `allFunctionCheckers` to be `functionCheckers + allMemberDeclarationCheckers` instead?

Reply:
The problem is currently `FirFunctionChecker` checks `FirFunction`, which may not be a member, (an anonymous function or error function). Currently, I don't know why the checker is not applied on anonymous function. Do we want to expand the scope for `FirFunctionChecker` to only check member declarations?

Reply:
What _would_ happen if we ran `allMemberDeclarationCheckers` on non-member functions?

Reply:
Well, that would require a type cast, and I think the resulted code would deviate pretty significantly from the current intention. Also, the current code gen works by checking the class hierarchy and populate these "allXXX" declarations. I can change the code gen to populate all classes rather than immediate parents to handle the constructor case correctly.